### PR TITLE
only keep modules with no earlier history entry

### LINKED ISSUE #10514: changing always-on modules messes with history behaviour
If I alter the white balance, revert back to a history step before that alteration and then change another module, white balance pops onto the history stack again even though it already appears lower in the stack. See the following video for a demonstration:

https://user-images.githubusercontent.com/9555491/143853479-6bebeac3-a14d-4966-bf19-389f163080fe.mp4

3.7.0+1578~g186f3a441

Same behaviour with demosaic, input color profile, output color profile, raw black/white point. Basically all of the modules that are forced on.

### PR DESCRIPTION
**Pull request by elstoc**
Resolves #10514

### CODE REVIEW

File: src/develop/develop.c [hunk 1]
```diff
@@ -814,7 +814,19 @@ static void _dev_add_history_item_ext(dt_develop_t *dev, dt_iop_module_t *module
...
     // printf("removing obsoleted history item: %s\n", hist->module->op);
-    if(!hist->module->hide_enable_button && !hist->module->default_enabled)
+
+    //check if an earlier instance of the module exists
+    gboolean earlier_entry = FALSE;
+    GList *prior_history = g_list_nth(dev->history, dev->history_end - 1);
+    while(prior_history)
+    {
+      dt_dev_history_item_t *prior_hist = (dt_dev_history_item_t *)(prior_history->data);
+      if(prior_hist->module->so == hist->module->so) earlier_entry = TRUE;
```
Comment (ralfbrown) [inline src/develop/develop.c:824]: You can break out of the loop after setting `earlier_entry`.

Comment (elstoc) [reply src/develop/develop.c:824]: Indeed. Thanks

**Review by TurboGit — APPROVED** [review]
Works for me, a nice corner case that I never experienced, good to see this fixed.

**Comment by elstoc** [comment]
Only saw it because I was playing around with custom white balance and trying to reproduce the other bug I fixed!

**Pull request merged**